### PR TITLE
Sanity Framework.

### PR DESCRIPTION
**Pull request by MACIEKBAKI**
## About the Pull Request
What it says on the tin. Did not touch medicine (antidepressants etc) because holy SHIT that is messy
## Changelog

:cl:
code: Framework for sanity, stored in the brain.
/:cl:

### CODE REVIEW

**Comment by MACIEKBAKI** [comment]
does: -nothing
affected by: -nothing

**Comment by Tsurupeta** [comment]
Wow, this is so barebones, make a codersprite of mood and add it to the interface so that person can see their "sanity" in separate health doll (which will be redrawn in new interface)

**Comment by aidan-bowman** [comment]
Forgive me if I'm wrong, but couldn't the variable sanity message on examine be bypassed by just spam shift-clicking someone?

**Comment by MACIEKBAKI** [comment]
> Forgive me if I'm wrong, but couldn't the variable sanity message on examine be bypassed by just spam shift-clicking someone?

It is, in fact, possible - probably optimal, too.
I am just coding this based off of pseudocode kaga gave me for how they envision it working

**Comment by MACIEKBAKI** [comment]
basically finished; effects will be implemented soon:tm:

**Comment by MACIEKBAKI** [comment]
for the record, work is on hold until I'm better after eye surgery

**Comment by MACIEKBAKI** [comment]
frankly waiting for review

File: code/modules/species/species_hud.dm [hunk 1]
```diff
@@ -6,12 +6,13 @@
 	var/has_pressure = 1  // Draw the pressure indicator.
 	var/has_nutrition = 1 // Draw the nutrition indicator.
 	var/has_bodytemp = 1  // Draw the bodytemp indicator.
+	var/has_sanity = 1 	  // Draw the sanity indicator.
```
Comment (Tsurupeta) [inline code/modules/species/species_hud.dm]:
```suggestion
	var/has_sanity = 0 	  // Draw the sanity indicator.
```

**Review by Tsurupeta — APPROVED** [review]

**Pull request merged**